### PR TITLE
Update process-agent check to use datadog-agent.yaml

### PR DESCRIPTION
**Pull request by conorbranagan**
### What does this PR do?

In beta9/5.22.0 the process-agent was given support for using the `datadog.yaml` file directly. This change brings this back to the Agent 6 side so we can simplify the setup flow for the Agent.

### CODE REVIEW

**Review by shang-wang — DISMISSED** [review]

**Review by olivielpeau — DISMISSED** [review]
Looks good, thanks for updating the docs too!

This will require an update on our `datadog` cookbook too, let me know if you can take care of it (or not, I have a task to update it too): see in particular https://github.com/DataDog/chef-datadog/blob/v2.13.0/recipes/_agent6_config.rb#L28

**Comment by olivielpeau** [comment]
actually, can you also remove this file? https://github.com/DataDog/datadog-agent/blob/6.0.0-beta.9/cmd/agent/dist/process-agent.conf

File: Dockerfiles/agent/Dockerfile
Comment (olivielpeau) [inline Dockerfiles/agent/Dockerfile]: this used to set `enabled: true` in the legacy config file. Is the new default behavior in the containerized agent to leave the setting to `false`?

**Review by olivielpeau — DISMISSED** [review]
LGTM!

Left a question, not a blocker to merge.

Should just need an approval from @DataDog/kenafeh on the small change in the Dockerfile.

File: Dockerfiles/agent/Dockerfile
Comment (xvello) [reply Dockerfiles/agent/Dockerfile]: If they read `DD_PROCESS_AGENT_ENABLED` through the `pkg/config` packet, we can just add this to YES by default in the dockerfile's env

Comment (conorbranagan) [reply Dockerfiles/agent/Dockerfile]: I'm actually not sure the history here but process collection isn't default anywhere in the old Agent so I don't think we need that here either. It will still start by default but it will only collect containers (this is the "false" case in agent 5 and 6)

**Review by olivielpeau — APPROVED** [review]
LGTM!

@DataDog/kenafeh: just waiting on your approval and we can merge

File: Dockerfiles/agent/Dockerfile
Comment (xvello) [reply Dockerfiles/agent/Dockerfile]: Okay, makes sense! I though this variable also disabled container collection.

LGTM then!

**Review by xvello — APPROVED** [review]

**Pull request merged**
